hack/generator/pkg/astmodel/resource_type.go

```diff
@@ -284,32 +370,34 @@ func (resource *ResourceType) RequiredPackageReferences() *PackageReferenceSet {
 
 // AsDeclarations converts the resource type to a set of go declarations
 func (resource *ResourceType) AsDeclarations(codeGenerationContext *CodeGenerationContext, declContext DeclarationContext) []dst.Decl {
-	packageName := codeGenerationContext.MustGetImportedPackageName(MetaV1PackageReference)
-
-	typeMetaField := defineField("", dst.NewIdent(fmt.Sprintf("%s.TypeMeta", packageName)), "`json:\",inline\"`")
-	objectMetaField := defineField("", dst.NewIdent(fmt.Sprintf("%s.ObjectMeta", packageName)), "`json:\"metadata,omitempty\"`")
-
 	/*
 		start off with:
 			metav1.TypeMeta   `json:",inline"`
 			metav1.ObjectMeta `json:"metadata,omitempty"`
 
 		then the Spec/Status properties
 	*/
-	fields := []*dst.Field{
-		typeMetaField,
-		objectMetaField,
-		defineField("Spec", resource.spec.AsType(codeGenerationContext), "`json:\"spec,omitempty\"`"),
+	var fields []*dst.Field
+	for _, property := range resource.EmbeddedProperties() {
+		f := property.AsField(codeGenerationContext)
+		if f != nil {
+			fields = append(fields, f)
+		}
 	}
 
-	if resource.status != nil {
-		statusType := resource.status.AsType(codeGenerationContext)
-		// ErroredTypes can be present as status but might not generate an actual status type
-		if statusType != nil {
-			fields = append(fields, defineField("Status", statusType, "`json:\"status,omitempty\"`"))
+	for _, property := range resource.Properties() {
+		f := property.AsField(codeGenerationContext)
+		if f != nil {
+			fields = append(fields, f)
 		}
 	}
 
+	if len(fields) > 0 {
+		// if first field has Before:EmptyLine decoration, switch it to NewLine
```

Comment:
Is this comment right? Should the code be checking for that decoration somewhere?

Reply:
Comment originated in code I copied from `ObjectType` - I've ammended both to make them more accurate.